node/pegnet/metadata.go

```diff
@@ -0,0 +1,35 @@
+package pegnet
+
+import (
+	"context"
+	"encoding/binary"
+)
+
+const createTableMetadata = `CREATE TABLE IF NOT EXISTS "pn_metadata" (
+	"name" TEXT NOT NULL,
+	"value" BLOB,
+	
+	UNIQUE("name")
+);
+`
+
+func (p *Pegnet) InsertSynced(ctx context.Context, height uint32) error {
+	b := make([]byte, 4)
+	binary.BigEndian.PutUint32(b, height)
+	_, err := p.DB.ExecContext(ctx, "REPLACE INTO pn_metadata (name, value) VALUES ($1, $2)", "synced", b)
```

Comment:
I think we should just make a `Synced struct` and json marshal it to the db. I could see us wanting to extend this later, and being able to add fields would make backward compatibility easier. 

We should eat the extra cost of storage now, to give us easier extendibility later.

Reply:
the metadata supports a bunch of key-value options. if we want extendability, we can just read the entire "metadata" table into a map[string]string

doing it this way allows us to use "synced" in queries

Reply:
since we can't query it anyway might as well make it a struct.

i moved the BlockSync struct to pegnet.BlockSync, which atm only contains Synced. if we add more records we should specify the "omitempty" tag for backward compatibility

Reply:
I was unaware you were trying to make it a queryable field. 

As long as it's a blob, json might remove headaches later :+1: